src/Expensify.js

```diff
@@ -102,28 +100,32 @@ class Expensify extends PureComponent {
     componentDidUpdate(prevProps) {
         const previousAccountID = lodashGet(prevProps, 'session.accountID', null);
         const currentAccountID = lodashGet(this.props, 'session.accountID', null);
+
         if (currentAccountID && (currentAccountID !== previousAccountID)) {
             PushNotification.register(currentAccountID);
         }
 
-        // If we previously had no authToken and now have an authToken we'll want to reshow the boot splash screen so
-        // that we can remove it again once the content is ready
-        const previousAuthToken = lodashGet(prevProps, 'session.authToken', null);
-        if (this.getAuthToken() && !previousAuthToken) {
-            this.showSplash();
-        }
+        if (this.state.isNavigationReady && this.state.isSplashShown) {
+            const authStackReady = this.props.initialReportDataLoaded && this.props.isSidebarLoaded;
+            const shouldHideSplash = !this.isAuthenticated() || authStackReady;
 
-        if (this.getAuthToken() && this.props.initialReportDataLoaded && this.props.isSidebarLoaded) {
-            this.hideSplash();
+            if (shouldHideSplash) {
+                this.hideSplash();
```

Comment:
Since `hideSplash()` is only called once I don't think it needs to be its own separate function anymore.